### ISSUE TITLE
Tentacle updater does not work with domain account

### ISSUE THREAD

**Issue by tothegills** [OctopusDeploy/Issues#2023]
It looks like the account has permissions but the install fails.  The task also reports success when it has failed.

See: https://secure.helpscout.net/conversation/122032270/1733/

**Comment by vanessalove** [comment]
The domain account needs to have permissions to run MSIs. This cannot be fixed from our end.

**Comment by tonii75** [comment]
I'm surprised this has been closed so quickly. I'm having the same problem currently and the only solution I have is to manually upgrade tentacles on all servers.
It seems to me a solution on your end could be for octopus tentacles to have a separate service that runs as local system for updating.

**Comment by tynor88** [comment]
Yes. This is not an issue with the domain account not having permissions. The domain account in our case is in the local administrator group on the particular server.

**Comment by tothegills** [comment]
Have tried with the Tentacle running as a domain user in the local administrators group and the Tentacle does not update.

**Comment by poulin69** [comment]
I have the same issue. I can execute the MSI-package locally on the server when logged on (runas prompt) as the account running the service which means it's not a permission problem

**Comment by carlpett** [comment]
We're also seeing this, not only is the upgrade failing, but it is actually rendering the tentacle useless. The service cannot be started again after attempting an upgrade, complaining about missing (unspecified) files. I had to uninstall and reinstall to get it working (msi repair did not work)

**Comment by poulin69** [comment]
How far are you on this issue? 

It is a bit pain to manually update the tentacle on every server

**Comment by Kemyke** [comment]
We have the same problem. So we are also waiting for this fix. Thanks!

**Comment by lonevvolf** [comment]
Same here.

**Comment by tothegills** [comment]
If I log on to a box as the AD user, install the Tentacle, and then run the Tentacle auto-updater it seems to work.

**Comment by tothegills** [comment]
Helpful exception when trying to run msiexec:
```
Error: System.ComponentModel.Win32Exception (0x80004005): Unknown error (0xfffffffe)
   at System.Diagnostics.Process.StartWithShellExecuteEx(ProcessStartInfo startInfo)
   at System.Diagnostics.Process.Start(ProcessStartInfo startInfo)
   at Octopus.Upgrader.SoftwareInstaller.Install(String msiPath) in Y:\work\refs\heads\master\source\Octopus.Upgrader\SoftwareInstaller.cs:line 16
   at Octopus.Upgrader.Program.Main(String[] args) in Y:\work\refs\heads\master\source\Octopus.Upgrader\Program.cs:line 39
```

**Comment by tothegills** [comment]
Release note: Tentacle automatic update now works when Tentacle is running as an Active Directory user

**Comment by poulin69** [comment]
Hi - where do you find that release note? I've now upgraded the server to 3.1.5 and still cannot upgrade tentacles running as an active diretory user account from the console

**Comment by poulin69** [comment]
Hi Shane

I’m not sure our problem is solved.

As I’ve tested and wrote to the case this morning, tentacles running as an active directory user account still cannot be upgraded from console I version 3.1.5.

Med venlig hilsen

Niels Poulin
Systemadministrator, D&S - Core Services

Københavns Universitet
Koncern-it
Nørregade 10, Postboks 2177
1017 København K

MOB 29 35 65 67
niels.poulin@adm.ku.dk<mailto:niels.poulin@adm.ku.dk>
it.ku.dk<http://it.ku.dk/>

[KU]

**Comment by alfhenrik** [comment]
Hi Niels,

The fix for this will be included in the next release `3.1.6`.

Thank you,
Henrik

**Comment by poulin69** [comment]
Hi Henrik

I’m looking forward to that – thank you

Med venlig hilsen

Niels Poulin
Systemadministrator, D&S - Core Services

Københavns Universitet
Koncern-it
Nørregade 10, Postboks 2177
1017 København K

MOB 29 35 65 67
niels.poulin@adm.ku.dk<mailto:niels.poulin@adm.ku.dk>
it.ku.dk<http://it.ku.dk/>

[KU]
